STYLE.md

```diff
@@ -0,0 +1,32 @@
+# Coding style
+
+* The Envoy source code is formated using clang-format. Thus all white space, etc.
+  issues are taken care of automatically. The Travis tests will automatically check
+  the code format and fail. There are make targets that can both check the format
+  (check_format) as well as fix the code format for you (fix_format).
+* Beyond code formatting, for the most part Envoy uses the
+  [Google C++ style guidelines](https://google.github.io/styleguide/cppguide.html).
```

Comment:
Do you also want to point to the Google style guides for Python, bash and BUILD files for completeness?

https://google.github.io/styleguide/pyguide.html
https://google.github.io/styleguide/shell.xml
https://github.com/bazelbuild/bazel/blob/master/site/versions/master/docs/skylark/build-style.md